Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js

```diff
@@ -0,0 +1,240 @@
+        this._viewportRow = new WI.DetailsSectionSimpleRow(WI.UIString("Viewport"));
+        this._framesRow = new WI.DetailsSectionSimpleRow(WI.UIString("Frames"));
+        this._resolutionRow = new WI.DetailsSectionSimpleRow(WI.UIString("Resolution"));
+        this._videoFormatRow = new WI.DetailsSectionSimpleRow(WI.UIString("Video Format"));
+        this._audioFormatRow = new WI.DetailsSectionSimpleRow(WI.UIString("Audio Format"));
+
+        let generalGroup = new WI.DetailsSectionGroup([this._sourceRow, this._viewportRow, this._framesRow, this._resolutionRow, this._videoFormatRow, this._audioFormatRow
+                                                       ]);
+        this._generalSection = new WI.DetailsSection("media-details-general", WI.UIString("General"), [generalGroup]);
+        this.element.appendChild(this._generalSection.element);
+
+        this._videoCodecRow = new WI.DetailsSectionSimpleRow(WI.UIString("Video Codec"));
+        this._transferRow = new WI.DetailsSectionSimpleRow(WI.UIString("Transfer Function"));
+        this._primariesRow = new WI.DetailsSectionSimpleRow(WI.UIString("Color Primaries"));
+        this._matrixRow = new WI.DetailsSectionSimpleRow(WI.UIString("Matrix Coefficients"));
+        this._fullRangeRow = new WI.DetailsSectionSimpleRow(WI.UIString("Color Range"));
+
+        let videoGroup = new WI.DetailsSectionGroup([this._videoCodecRow, this._primariesRow, this._transferRow, this._matrixRow, this._fullRangeRow]);
+        this._videoSection = new WI.DetailsSection("media-video-details", WI.UIString("Video Details"), [videoGroup]);
+        this.element.appendChild(this._videoSection.element);
+
+        this._audioCodecRow = new WI.DetailsSectionSimpleRow(WI.UIString("Audio Codec"));
+        this._sampleRateRow = new WI.DetailsSectionSimpleRow(WI.UIString("Sample Rate"));
+        this._channelsRow = new WI.DetailsSectionSimpleRow(WI.UIString("Channels"));
+
+        let audioGroup = new WI.DetailsSectionGroup([this._audioCodecRow, this._sampleRateRow, this._channelsRow]);
+        this._audioSection = new WI.DetailsSection("media-audio-details", WI.UIString("Audio Details"), [audioGroup]);
+        this.element.appendChild(this._audioSection.element);
+    }
+
+    async layout()
+    {
+        super.layout();
+        
+        let stats = await this.domNode.getMediaStats();
+        this.source = stats.source;
+        this.viewport = stats.viewport;
+        this.devicePixelRatio = stats.devicePixelRatio;
+        this.quality = stats.quality;
+        this.videoFormat = stats.videoFormat;
+        this.audioFormat = stats.audioFormat;
+        this.video = stats.video;
+        this.audio = stats.audio;
+    }
+
+    get source() {
+        return this._source;
+    }
+
+    set source(source) {
+        if (this._source === source)
+            return;
+
+        this._source = source;
+        this._sourceRow.value = WI.UIString(source);
```

Comment:
This isn't a valid way to express a UIString in Web Inspector. The localized string file is generated using the `update-webkit-localizable-strings`, and this will not produce valid results. Localized strings should be thought of as constants. For something like this you will want to have a switch-case that does the conversion from protocol values to a localized string.